src/node_url_pattern.cc

```diff
@@ -461,9 +413,15 @@ URLPattern::URLPatternOptions::FromJsObject(Environment* env,
   if (obj->Get(env->context(), env->ignore_case_string())
           .ToLocal(&ignore_case)) {
     if (!ignore_case->IsBoolean()) {
+      THROW_ERR_INVALID_ARG_TYPE(env, "options.ignoreCase must be a boolean");
       return std::nullopt;
     }
     options.ignore_case = ignore_case->IsTrue();
+  } else {
+    // If ToLocal returns false, the assumption is that getting the
+    // ignore_case_string threw an error, let's propagate that now
+    // by returning std::nullopt.
+    return std::nullopt;
```

Comment:
On line 416 we throw an error, whereas in here we don't. I think we should throw an error here as well. If not, it won't be possible if std::null opt has thrown or not in the caller function

Reply:
No because an error at this point would have *already* been scheduled as the added test demonstrates and comment explains.